internal/controller/customreplicaset_controller.go

```diff
@@ -336,215 +359,223 @@ func (r *CustomReplicaSetReconciler) findChildPods(ctx context.Context, req ctrl
 	return childPods, nil
 }
 
-func (r *CustomReplicaSetReconciler) managePods(ctx context.Context, totalAvailablePods, availableStdPods, availableUpgPods int, crs *customreplicasetv1.CustomReplicaSet, childPods corev1.PodList, log logr.Logger) error {
-	if totalAvailablePods < int(crs.Spec.Replicas) {
-		if err := r.createPods(ctx, totalAvailablePods, availableStdPods, availableUpgPods, crs, log); err != nil {
-			log.Error(err, "Failed to create pods")
-			return err
-		}
-	} else if totalAvailablePods > int(crs.Spec.Replicas) {
-		if err := r.deletePods(ctx, totalAvailablePods, availableStdPods, availableUpgPods, crs, childPods, log); err != nil {
-			log.Error(err, "Failed to delete pods")
-			return err
-		}
-	} else {
-		if err := r.upgradeOrDowngradePods(ctx, availableStdPods, availableUpgPods, crs, childPods, log); err != nil {
-			log.Error(err, "Failed to upgrade or downgrade pods")
-			return err
-		}
-	}
-	return nil
-}
-
-func (r *CustomReplicaSetReconciler) createPods(ctx context.Context, totalPods, stdPods, upgPods int, cr *customreplicasetv1.CustomReplicaSet, log logr.Logger) error {
-	for totalPods < int(cr.Spec.Replicas) {
-		var newPod *corev1.Pod
-		// Create any missing upgraded pods first
-		if upgPods < int(cr.Spec.Partition) {
-			newPod = r.newPodForCR(cr, true)
-			upgPods++
-		} else {
-			newPod = r.newPodForCR(cr, false)
-			stdPods++
-		}
-
-		if err := r.Create(ctx, newPod); err != nil {
-			log.Error(err, "Unable to create new pod")
-			return err
+func (r *CustomReplicaSetReconciler) managePods(ctx context.Context, crs *customreplicasetv1.CustomReplicaSet, childPods corev1.PodList, podsPerRevision map[int][]*corev1.Pod, totalAvailPods int, latestRevision *v1.ControllerRevision) error {
+	latestPods, ok := podsPerRevision[int(latestRevision.Revision)]
+	if !ok {
+		fmt.Println("No pods found of latest revision")
+		latestPods = []*corev1.Pod{}
+	}
+	if totalAvailPods < int(crs.Spec.Replicas) {
+		podsToCreate := int(crs.Spec.Replicas) - totalAvailPods
+		newPodsToCreate := int(crs.Spec.Partition) - len(latestPods)
```

Comment:
For sanity purposes, you should ensure that the created pod number is capped by the podToCreate (in case user puts a wrong big Partition value). I understand you can add a validationwebhook to prevent this from happening, but you should add a check since the webhook is not there yet.

Reply:
Yes your right I put a cap, where partition can now be at max equal to replicas value